mono/metadata/icall.c

```diff
@@ -8526,10 +8526,14 @@ ves_icall_System_IO_LogcatTextWriter_Log (const char *appname, gint32 level, con
 static MonoIcallTableCallbacks icall_table;
 static mono_mutex_t icall_mutex;
 static GHashTable *icall_hash = NULL;
-static GHashTable *icall_hash_foreign = NULL;
 static GHashTable *jit_icall_hash_name = NULL;
 static GHashTable *jit_icall_hash_addr = NULL;
 
+typedef struct _MonoIcallHashTableValue {
+	gconstpointer method;
```

Comment:
If `const` is only going to be frequently cast away, consider removing it here.

Reply:
hopefully it won't be needed at more places since any new usage should use gconstpointer, reason that it needs to be casted away was an old API, but that doesn't seem to be exposed so maybe I can keep it const a little longer. Will update.